code/__HELPERS/unsorted.dm

```diff
@@ -174,59 +174,74 @@ Turf and target are seperate in case you want to teleport some distance from a t
 
 
 
-//This will update a mob's name, real_name, mind.name, data_core records, pda and id
-//Calling this proc without an oldname will only update the mob and skip updating the pda, id and records ~Carn
-/mob/proc/fully_replace_character_name(var/oldname,var/newname)
-	if(!newname)	return 0
+/**
+ * This will update a mob's name, real_name, mind.name, data_core records, pda and id.
+ * Calling this proc without an oldname will only update the mob and skip updating the pda, id and records. ~Carn
+ */
+/mob/proc/fully_replace_character_name(oldname, newname)
+	if (!newname)
+		return 0
+
 	real_name = newname
+
 	name = newname
-	if(mind)
+
+	if (mind)
 		mind.name = newname
-	if(dna)
+
+	if (dna)
 		dna.real_name = real_name
 
-	if(oldname)
-		//update the datacore records! This is goig to be a bit costly.
-		for(var/list/L in list(data_core.general,data_core.medical,data_core.security,data_core.locked))
-			var/datum/data/record/R = find_record("name", oldname, L)
+	if (oldname)
+		/*
+		 * Update the datacore records!
+		 * This is going to be a bit costly.
+		 */
+		for (var/list/L in list(data_core.general, data_core.medical, data_core.security,data_core.locked))
+			if (L)
+				var/datum/data/record/R = find_record("name", oldname, L)
+
+				if (R)
+					R.fields["name"] = newname
 
-			if(R)
-				R.fields["name"] = newname
+		// update our pda and id if we have them on our person
+		var/search_id = TRUE
 
-		//update our pda and id if we have them on our person
-		var/list/searching = GetAllContents(searchDepth = 3)
-		var/search_id = 1
-		var/search_pda = 1
+		var/search_pda = TRUE
 
-		for(var/A in searching)
-			if( search_id && istype(A,/obj/item/weapon/card/id) )
-				var/obj/item/weapon/card/id/ID = A
-				if(ID.registered_name == oldname)
+		for (var/object in get_contents_in_object(src, /atom/movable))
+			if (search_id && istype(object, /obj/item/weapon/card/id))
+				var/obj/item/weapon/card/id/ID = object
+
+				if (ID.registered_name == oldname)
 					ID.registered_name = newname
 					ID.name = "[newname]'s ID Card ([ID.assignment])"
-					if(!search_pda)	break
-					search_id = 0
 
-			else if( search_pda && istype(A,/obj/item/device/pda) )
-				var/obj/item/device/pda/PDA = A
-				if(PDA.owner == oldname)
+					if (!search_pda)
+						break
+
+					search_id = FALSE
+			else if (search_pda && istype(object, /obj/item/device/pda))
+				var/obj/item/device/pda/PDA = object
+
+				if (PDA.owner == oldname)
 					PDA.owner = newname
 					PDA.name = "PDA-[newname] ([PDA.ownjob])"
-					if(!search_id)	break
-					search_pda = 0
-
-		//Fixes renames not being reflected in objective text
-		var/list/O = (typesof(/datum/objective)  - /datum/objective)
-		var/length
-		var/pos
-		for(var/datum/objective/objective in O)

```

Comment:
That code doesn't even work.
Paths have no type, so Byond skips them in type loops.